tasmota/berry/drivers/tm1637.be

```diff
@@ -0,0 +1,190 @@
+        gpio.digital_write(self.PIN_DIO, 0)
+        gpio.digital_write(self.PIN_CLK, 1)
+        gpio.digital_write(self.PIN_DIO, 1)
+    end
+
+    def ack()
+        gpio.digital_write(self.PIN_CLK, 0)
+        gpio.pin_mode(self.PIN_DIO, gpio.INPUT_PULLUP)
+        var ack_state = gpio.digital_read(self.PIN_DIO) == 0
+        gpio.digital_write(self.PIN_CLK, 1)
+        gpio.digital_write(self.PIN_CLK, 0)
+        gpio.pin_mode(self.PIN_DIO, gpio.OUTPUT)
+        return ack_state
+    end
+
+    def write_bit(bitval)
+        gpio.digital_write(self.PIN_CLK, 0)
+        gpio.digital_write(self.PIN_DIO, bitval)
+        gpio.digital_write(self.PIN_CLK, 1)
+    end
+
+    def write_byte(byteval)
+        for pos: 0..7
+            self.write_bit((byteval >> pos) & 0x01)
+        end
+    end
+
+    def send_command(command)
+        self.start()
+        self.write_byte(command)
+        var ack_state = self.ack()
+        self.stop()
+        return ack_state
+    end
+
+    def send_data(data)
+        var ack_state = true
+        self.start()
+        for i : 0..size(data)-1
+            self.write_byte(data[i])
+            ack_state = self.ack() && ack_state
+        end
+        self.stop()
+        return ack_state
+    end
+
+    #  0-8 range, 0 to 'OFF'
+    def set_on(brightness)
+        if brightness == nil || brightness > 8
+            brightness = 8
+        elif brightness < 0
+            brightness = 0
+        end
+        var cmd = self.CMD_CTRL
+        if brightness
+            cmd |= self.CMD_DISP_ON
+            brightness -= 1
+        end
+        return self.send_command(cmd | brightness)
+    end
+
+    def print(num)
+        import string
+
+        num = str(num)
+        var max_str_len = 4
+
+        do
+            var dot_pos = string.find(num, '.')
+            if dot_pos >= 0 && dot_pos < 5
+                max_str_len = 5
+            end
+        end
+        if size(num) > max_str_len
+            num = string.split(num, max_str_len)[0]
+        end
+        num = string.format('%4s', num)
+        var payload = bytes(-5)
+        payload[0] = self.CMD_ADDR
+        var int_offset = 1
+        for i : 0..size(num)-1
+            if num[i] == '.'
+                payload[i] |= self.SYMB_DOT
+                int_offset = 0
+            else
+                payload[i + int_offset] = self.DIGIT_MAP[num[i]]
+            end
+        end
+        var ack_state = self.send_command(self.CMD_DATA) && self.send_data(payload)
+        if !ack_state
+            log('TM1637 - no ACK, please check connections')
+        end
+        return ack_state
+    end
+
+    def clear()
+        self.print('    ')
+    end
+
+    # Won't be called on the system restart
+    def deinit()
+        self.set_on(0)
+    end
+end
+
+def tm_add_custom_commands(dio, clk)
+    var tm = Tm1637(dio, clk)
+    tm.clear()
+    tm.set_on(4)
+    tasmota.add_cmd('tm_print', def(cmd, idx, payload)
```

Comment:
Tasmota commands usually don't have an underscore. Can you reconsider for consistency with other commands?

Reply:
Sure, done